src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestStream.cs

```diff
@@ -122,11 +144,11 @@ public override int Read(byte[] buffer, int offset, int count)
             throw new NotSupportedException();
         }
 
-        internal void EndUpload()
+        internal async Task EndUploadAsync(CancellationToken token)
```

Comment:
It looks like this is only called from one place.  It's so short an implementation, any reason not to just inline it into the caller?  There's some overhead associated with each async method invocation, especially it contains an await that yields, so from an efficiency perspective it's better to have fewer async method invocations.  That's sometimes is at odds with a desire to refactor into small reusable methods, but in this case, it doesn't appear to be reusable nor is it particularly verbose.